experimental/veggieseasons/lib/screens/list.dart

```diff
@@ -32,14 +32,14 @@ class ListScreen extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final themeData = CupertinoTheme.of(context);
```

Comment:
This is the reason the "In Season Today" font is not re-rendering with the correct color when you switch between dark and light mode. I'll explain:

When you call one of these `of` methods on an InheritedWidget and give it a context, the InheritedWidget remembers the context you give it, and will automatically rebuild the widget at that point in the tree if it needs to (such as when CupertinoTheme detects that the device has switched from dark to light mode or vice versa).

In this bit of code, you're using the context of ListScreen itself. That's fine for most cases, but here, ListScreen is building a CupertinoTabView. CupertinoTabView has as one of its children, a Navigator widget.

CupertinoTabView gives that Navigator the builder method on line 38 to use when building its children. If you rebuild the CupertinoTabView and give it a *different* builder function, the Navigator will use that new function any time it needs to build its child widgets in the future, **but** it will not immediately trigger a rebuild of them. This is because Navigators need to hold multiple routes simultaneously, stacked on top of each other. They need to remember what routes have already been pushed. Those routes have their own widgets and state, which could be lost if they were automatically rebuilt in response to the new builder method.

This can be slightly confusing to wrap your head around. Fortunately, the fix is easy: move the call to the InheritedWidget to be *inside* the builder method that the Navigator is using. That way, the context that's used will *also* be inside the Navigator's subtree of widgets, which means there's no change to the builder method, and instead the right widget will be rebuilt automatically, further down the tree.

In this case, that means taking the call that's now on line 35, and move it back to line 42. The font will begin to change correctly after that.

Reply:
That makes a lot of sense. Thanks for the insight!